docs/overview.rst

```diff
@@ -0,0 +1,67 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+
+Overview
+========
+
+In order to upload files, a Location needs to be defined first.
```

Comment:
I would add an extra gentle introduction paragraph, something like (please improve!):

> Invenio-Files-REST is a files storage module. You can store and retrieve files in a similar way to Amazon S3 APIs.
> 
> To understand how to safely manage files in Invenio, this guide will introduce you to the key concepts and terminology that you need to understand.

I will, IMHO, but please let's understand together what is best, start from "the end", files: this is because if I am  new to these concepts, I don't know/understand why I need locations and buckets.
I would try to answer this question: I have a file, why would I need a location/bucket/object version?
I would also do **not** focus on upload/download only, because this if for HTTP. We can use the module programmatically.

The text below should be improved... but you should get my point.

> Files are represented by ``FileInstance``,  which describes the path (or location???) the file on disk. Each file belongs to a ``Bucket``, a container of files. Buckets are important to organize your storage or, for example, to define quotas. Since a file can have a complex set of metadata, for example name, type, size, etc., an ``ObjectVersion`` is associated to each file.

And then, maybe start the detailed description of each object.


> Locations
> ----------
> 
> A Location is ...